src/jit/morph.cpp

```diff
@@ -7722,7 +7722,7 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
     {       
         GenTreePtr value = gtArgEntryByArgNum(call, 2)->node;
 
-        if (value->OperGet() == GT_CNS_INT && value->AsIntConCommon()->IconValue() == 0)
+        if (value->IsZero())
```

Comment:
I think that the easiest way forward would be just to add just these four methods:
IsIntOne
IsIntegralOne
IsIntMinusOne
IsIntegralMinusOne

Is that a satisfactory way forward?